sound/soc/codecs/rt711.c

```diff
@@ -405,10 +405,13 @@ static int rt711_set_jack_detect(struct snd_soc_component *component,
 	struct snd_soc_jack *hs_jack, void *data)
 {
 	struct rt711_priv *rt711 = snd_soc_component_get_drvdata(component);
+	struct snd_soc_dapm_context *dapm =
+		snd_soc_component_get_dapm(rt711->component);
 
 	while (!rt711->hw_init)
 		return -EAGAIN;
 
+	snd_soc_dapm_mutex_lock(dapm);
```

Comment:
for my education, why is this lock needed?

Reply:
I checked all your comments. Thanks for your suggestions.
I will modify it tomorrow.
For this question, I want to make sure the calibration is done without other factors.
If the machine driver calls snd_soc_component_set_jack API, the calibration work and rt711_set_jack_detect will be interleaved.

Reply:
ok, then a comment would definitively help reviewers if we push this upstream